pkg/venafi/cloud/connector.go

```diff
@@ -792,33 +792,19 @@ func (c *Connector) ProvisionCertificate(req *endpoint.ProvisioningRequest, opti
 		}
 	}
 
-	graphqlClient := c.getGraphqlClient()
-
-	ctx := context.Background()
-
 	// Getting Keystore to find type
 	keystoreIDInput := util.StringPointerToString(reqData.KeystoreID)
 	keystoreNameInput := util.StringPointerToString(reqData.KeystoreName)
 	providerNameInput := util.StringPointerToString(reqData.ProviderName)
 
 	log.Printf("fetching keystore information for provided keystore information. KeystoreID: %s, KeystoreName: %s, ProviderName: %s", keystoreIDInput, keystoreNameInput, providerNameInput)
-	data, err := cloudproviders.GetCloudKeystores(ctx, graphqlClient, req.KeystoreID, reqData.KeystoreName, nil, reqData.ProviderName)
+	ctx := context.Background()
+	cloudKeystore, err := c.cloudProvidersClient.GetCloudKeystore(ctx, req.KeystoreID, reqData.KeystoreName, reqData.ProviderName)
 	if err != nil {
 		return nil, err
 	}
 
-	if data == nil || data.CloudKeystores == nil {
-		return nil, fmt.Errorf("could not find keystore with provided information. KeystoreID: %s, KeystoreName: %s, ProviderName: %s", keystoreIDInput, keystoreNameInput, providerNameInput)
-	}
-
-	if len(data.CloudKeystores.Nodes) != 1 {
-		return nil, fmt.Errorf("could not find keystore with provided information. KeystoreID: %s, KeystoreName: %s, ProviderName: %s", keystoreIDInput, keystoreNameInput, providerNameInput)
-	}
-
-	// grabbing found keystore for later
-	cloudKeystore := data.CloudKeystores.Nodes[0]
-
-	keystoreIDString := cloudKeystore.GetId()
+	keystoreIDString := cloudKeystore.ID
```

Comment:
I'm wondering how the cloudKeystore.GetId() existed and why now no... it's always preferable to use an access method instead of refer directly to the property of an struct

Reply:
because the previous value was directly from the generated code. Now I'm using a web client, and currently the struct I'm using doesn't have those methods defined. I didn't see them necessary